### ISSUE TITLE
[html] Add support for Angular2

### ISSUE THREAD

**Issue by maliksahil** [Microsoft/vscode#4927]
Currently VSCode does not have support for Angular2. But it does seem to have prompts for Angular1.
This makes writing Angular2 code difficult.
For instance, when authoring a template for a component, if I wish to use *ngFor, I am prompted to pick Angular1 directives - 

![image](https://cloud.githubusercontent.com/assets/4992186/14260484/b3138436-fa79-11e5-81ca-7e0aa86220cf.png)

Also, VSCode is currently blind to Angular2 component semantics. It would be great if those showed up in intellisense, which would make composing applications a lot easier. This of course applies to more than just components.

I propose that support for Angular2 be added, and it should be implemented as an extension. So individual projects can pick and choose between Angular1 or Angular2.

**Comment by Tyriar** [comment]
I think this is blocked on https://github.com/Microsoft/vscode/issues/2784

**Comment by johnpapa** [comment]
I agree that better support would be awesome!

For now there are html and typescript snippets for angular2 that i wrote in the marketplace. https://marketplace.visualstudio.com/items?itemName=johnpapa.Angular2

Also, there is a request in user voice for auto imports. this would rock https://visualstudio.uservoice.com/forums/293070-visual-studio-code/suggestions/13344849-auto-imports-in-typescript

**Comment by aeschli** [comment]
It would be great if an extension could add that behavior. Extensions can already participate on code completions. They would have to do the same parsing as the HTML language mode to figure out at what location the user wants to complete.

**Comment by oodboo** [comment]
Intellisense, color schema and Emet already works in HTML, could you apply on template?
For instance in Angular 2 if :
     template:`  ` 
pattern is recognized everything inside of  `  `  should behave as HTML?

**Comment by aeschli** [comment]
@oodboo Is that in JavaScript? If yes, please file a separate issue for it. This bug is about improving the HTML support.

**Comment by oodboo** [comment]
I have just found #2000 which basically says the same thing ... so I do not know should I copy text there ...

**Comment by aeschli** [comment]
@oodboo #2000 looks like the right bug. Yes, add your example and suggestions there.

**Comment by jorgeas80** [comment]
So, no intellisense for angular 2 yet? I'm using the amazing @johnpapa snippets, but it would be great to have intellisense and autoimports for angular 2

**Comment by stgolem** [comment]
Look for info here https://github.com/microsoft/typescript/issues/6508 and here https://github.com/angular/angular/issues/7482

**Comment by mgechev** [comment]
Yes, https://github.com/angular/angular/issues/7482 will provide great improvements for the Angular development experience in VSCode. Currently with the tslint plugin and [codelyzer](https://www.npmjs.com/package/codelyzer) you can get warnings in inline templates/styles, for instance, for binding to private/missing properties or unused CSS.

**Comment by kuncevic** [comment]
I am just creating angular 2 inline template It wold of being nice to autocorrelate dom events like if you type `<button (mouse...` expecting it will autocomplete with `mouseover`

**Comment by mgechev** [comment]
@kuncevic this is out of the scope of codelyzer but the language service that the Angular team is working on will provide such feature. Take a look at this talk by [Misko Hevery from ng-europe](https://youtu.be/wpxnU62mNJ4?t=13m47s). Soon should be released also a VSCode extension which works together with the language service in order to provide autocompletion and much more.

**Comment by alejandrohn** [comment]
I have vs code installed but the variables of the class are not displayed in the template (html) of component, not intellisense, not show the variables, I have use angular 4 with snippets of john papa, have I install some extension?

**Comment by kukjevov** [comment]
Here is extension which is still work in progress.

https://github.com/angular/vscode-ng-language-service

**Comment by alejandrohn** [comment]
Thanks, it really works

**Comment by mjbvz** [comment]
Yes, this extension provides angular 2 support: https://marketplace.visualstudio.com/items?itemName=Angular.ng-template
